tests/testtransportonrunning.nim

```diff
@@ -0,0 +1,172 @@
+{.used.}
+
+# Nim-LibP2P
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import chronos
+import ../libp2p/[
+  multiaddress, 
+  transports/tcptransport, 
+  transports/memorytransport,
+  transports/quictransport,
+  transports/wstransport,
+  transports/tortransport,
+  upgrademngrs/upgrade,
+  crypto/crypto
+]
+import ./helpers, ./stubs/torstub
+
+suite "Transport onRunning":
+  asyncTest "onRunning event is fired when transport starts":
+    let upgrader = Upgrade()
+    let transport = TcpTransport.new(upgrade = upgrader)
+    var eventFired = false
+    
+    proc onRunningHandler() {.async.} =
+      await transport.onRunning.wait()
+      eventFired = true
+    
+    asyncSpawn onRunningHandler()
+    
+    check:
+      eventFired == false
+      transport.running == false
+    
+    await transport.start(@[MultiAddress.init("/ip4/127.0.0.1/tcp/0").tryGet()])
+    
+    # Give the event handler time to run
+    await sleepAsync(10.milliseconds)
+    
+    check:
+      eventFired == true
+      transport.running == true
+    
+    await transport.stop()
+
+  asyncTest "onRunning event can be awaited":
+    let upgrader = Upgrade()
+    let transport = TcpTransport.new(upgrade = upgrader)
+    
+    let startFut = transport.start(@[MultiAddress.init("/ip4/127.0.0.1/tcp/0").tryGet()])
+    
+    await transport.onRunning.wait()
+    
+    check:
+      transport.running == true
+    
+    await startFut
+    await transport.stop()
+
+  asyncTest "onRunning works with memory transport":
+    let upgrader = Upgrade()
+    let transport = MemoryTransport.new(upgrade = upgrader)
+    var eventFired = false
+    
+    proc onRunningHandler() {.async.} =
+      await transport.onRunning.wait()
+      eventFired = true
+    
+    asyncSpawn onRunningHandler()
+    
+    await transport.start(@[])
+    
+    # Give the event handler time to run
+    await sleepAsync(10.milliseconds)
+    
+    check:
+      eventFired == true
+      transport.running == true
+    
+    await transport.stop()
+
+  asyncTest "onRunning works with quic transport":
```

Comment:
test kind reuse same code, could we create proc that will reuse common logic in tests?

for example could test be, like this:


```
asyncTest "onRunning with websocket":
   # setup transport here ...
   runTest(transport)
   
asyncTest "onRunning with tcptransport":
   # setup transport here ...
   runTest(transport)

... etc
```